src/main/java/com/froobworld/farmcontrol/controller/task/ActionPerformTask.java

```diff
@@ -57,9 +58,9 @@ public void run() {
             }
         }
         for (SnapshotEntity snapshotEntity : unTriggerActionMap.keySet()) {
-            Mob entity = snapshotEntity.getEntity();
+            Entity entity = snapshotEntity.getEntity();
             schedulerHook.runEntityTaskAsap(() -> {
-                if (!entity.isValid()) {
+                if (!(entity instanceof Mob mob) || !entity.isValid()) {
```

Comment:
I'd prefer if we got rid of this line and changed change undoAction to accept Entity, just in case it's needed for any future actions that might be added

Reply:
Done!